packages/x-date-pickers/src/internals/components/CalendarOrClockPicker/CalendarOrClockPicker.tsx

```diff
@@ -112,8 +112,7 @@ export function CalendarOrClockPicker<TDate, View extends CalendarOrClockPickerV
   const isLandscape = useIsLandscape(views, orientation);
   const wrapperVariant = React.useContext(WrapperVariantContext);
 
-  const toShowToolbar =
-    typeof showToolbar === 'undefined' ? wrapperVariant !== 'desktop' : showToolbar;
+  const toShowToolbar = showToolbar ?? wrapperVariant !== 'desktop';
```

Comment:
This is not related to the bug, just a small change to be coherent with the range pickers check.